dvc/dependency/base.py

```diff
@@ -30,5 +36,66 @@ class Dependency(Output):
     )  # type: Type[DvcException]
     IsStageFileError = DependencyIsStageFileError  # type: Type[DvcException]
 
-    def update(self, rev=None):
-        pass
+    def get_used_objs(
+        self, **kwargs
+    ) -> Dict[Optional["ObjectDB"], Set["HashInfo"]]:
+        from dvc_data.build import build
+        from dvc_data.objects.tree import Tree, TreeError
+
+        if not self.version_id:
+            return super().get_used_objs(**kwargs)
+
+        used_obj_ids: Dict[
+            Optional["ObjectDB"], Set["HashInfo"]
+        ] = defaultdict(set)
+        local_odb = self.repo.odb.local
+        try:
+            object_store, _, obj = build(
+                local_odb,
+                self.fs_path,
+                self.fs,
+                local_odb.fs.PARAM_CHECKSUM,
+            )
+        except (FileNotFoundError, TreeError) as exc:
+            raise DvcException(
+                f"The path '{self.fs_path}' does not exist in the remote."
+            ) from exc
+        # TODO: support versioned import writes (push)
+        object_store = copy(object_store)
+        object_store.read_only = True
+
+        used_obj_ids[object_store].add(obj.hash_info)
+        if isinstance(obj, Tree):
+            used_obj_ids[object_store].update(oid for _, _, oid in obj)
+        return used_obj_ids
+
+    def workspace_status(self):
+        if not self.version_id:
+            return super().workspace_status()
+
+        current = self.version_id
+        fs_path = self.fs.path.version_path(self.fs_path, None)
+        updated = self.fs.info(fs_path)["version_id"]
```

Comment:
For the record: discussed that we can capture this during object `build()` in `meta`. Also might want to add something like a `no_hash=True` flag to `build`, to avoid computing hashes.